### ISSUE TITLE
Static or Dynamic Masking Strategy

### ISSUE THREAD

**Issue by h19920918** [google-research/text-to-text-transfer-transformer#147]
Hi, I have a question for the masking strategy in the 'T5'.

In the Roberta, they pointed out static masking strategy used in the BERT, where a token is masked in the pre-processing step. (Masked token is never changed in the all training steps.)
So, they proposed dynamic masking strategy where masked tokens are changed for each training step.

I would like to know what kind of masking strategy is applied to this repository.
As I know, the authors of the 'T5' didn't describe about it.
Could you give to me advice about this?

**Comment by adarob** [comment]
We mask dynamically. However, our training set (C4) so large that is we don't repeat examples anyways!

**Comment by h19920918** [comment]
@adarob Thank you for your quickly answer.

I don't understand one more thing in your code.

In https://github.com/google-research/text-to-text-transfer-transformer/blob/4598ccbda22ba051e565b8d13326afe1cbd1de33/t5/data/preprocessors.py#L2115,

you describe as below:

We want to generate training examples like
  "We hold X to be Y that Z" -> "X these truths Y self evident Z that"

As I understand in the 'T5' paper,

In my assumption, its original sentence is "We hold these truths to be self evident that".
Following the 'T5' paper, it is converted as

Input: We hold X to be Y that
Target: X these truths Y self evident Z

I would like to know why the target has 'that'.
Additionally, in the language processing, there should be '(BOS)' and '(EOS)' tokens in the input and target.

In my best knowledge, the input and target will be

Input: We hold X to be Y that (EOS)
target: (BOS) X these truths Y self evident Z (EOS)

Is my assumption correct?
